### PR TITLE
Explain break/continue behavior

### LINKED ISSUE #5991: `trap` and `switch` are missing
* `trap` supports `continue` as well: the trapped error is _silently_ ignored and execution continues.

* `switch` supports `continue` as well: with a _scalar_ input, it effectively behaves like `break` (exits the statement as a whole), but with an _array-valued_ (collection) input, `continue` _moves on to the next array element_.

```powershell
# -> 1, 3
switch (1,2,3) {
  2 { continue }  # moves on to the next element, 3
  default { $_ }
}
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 070aa957-c96a-ec60-cb6a-7fe5f3bad820
* Version Independent ID: 898c8336-37f1-296d-768c-60397c723ba2
* Content: [about_Continue - PowerShell](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_continue?view=powershell-7.1)
* Content Source: [reference/7.1/Microsoft.PowerShell.Core/About/about_Continue.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/7.1/Microsoft.PowerShell.Core/About/about_Continue.md)
* Product: **powershell**
* GitHub Login: @SteveL-MSFT
* Microsoft Alias: **slee**

### LINKED ISSUE #5992: `trap` is missing as a statement to which `break` applies; `trap`-related example is flawed
* `trap` supports `break` as well and should be mentioned at the top: adding `break` effectively turns the error into a runspace-terminating one.

* Therefore, the following example isn't correct:

> You break out of the other looping statements in the same way you break out of the Foreach loop. In the following example, the Break statement exits a While statement when a DivideByZeroException exception is trapped using the Trap statement.
```powershell
$i = 3
while ($true) {
  trap [DivideByZeroException] {
    Write-Host 'divide by zero trapped'
    break
  }
   1 / $i--
}
```

The `break`, due to being inside the `trap` script block, exits the entire runspace - it is unrelated to the enclosing `while` loop.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 276062b7-b784-adb4-f78c-67cbe4e8f655
* Version Independent ID: e5c77926-c030-4723-9273-5415f1d9e723
* Content: [about_Break - PowerShell](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_Break?view=powershell-7.1)
* Content Source: [reference/7.1/Microsoft.PowerShell.Core/About/about_Break.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/7.1/Microsoft.PowerShell.Core/About/about_Break.md)
* Product: **powershell**
* GitHub Login: @SteveL-MSFT
* Microsoft Alias: **slee**

### LINKED ISSUE #5994: Warn of the pitfalls of `break` and `continue` uses outside constructs that directly support them
See https://github.com/PowerShell/PowerShell/issues/12742 for background.

When `break` and `continue` are used outside of a construct that directly supports these keywords (loops, `switch`, `trap`), PowerShell looks _up the call stack_ for an enclosing loop and _quietly terminates the runspace_ if no such loop is found.

This means that functions and scripts that inadvertently use a `break` or `continue` without an enclosing construct that supports it can end up inadvertently terminating their _callers_ as well.

The guidance that should be given to users is:

* Only ever use `break` and `continue` inside constructs that support it.

  * In particular, do _not_ use these keywords inside a pipeline: `break` inside a `ForEach-Object` script block does _not_ just exits the pipeline, it potentially terminates the entire runspace, as does `continue`. (There currently is no way to exit a pipeline prematurely, except if you wrap a _dummy loop_ around the pipeline for the `break` to break out of; see [this SO answer](https://stackoverflow.com/a/60341948/45375) and the feature request to directly support on-demand exit from a pipeline, https://github.com/PowerShell/PowerShell/issues/3821)

* The only potential exception is for writing reusable code explicitly designed and documented to affect the caller's flow control.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e4d6714f-5609-52c2-987c-b0522d038b2b
* Version Independent ID: e5c77926-c030-4723-9273-5415f1d9e723
* Content: [about_Break - PowerShell](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_break?view=powershell-7)
* Content Source: [reference/7.0/Microsoft.PowerShell.Core/About/about_Break.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/7.0/Microsoft.PowerShell.Core/About/about_Break.md)
* Product: **powershell**
* GitHub Login: @SteveL-MSFT
* Microsoft Alias: **slee**

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes [AB#1730586](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1730586) - Fixes #5991 - `trap` and `switch` are missing
Fixes [AB#1730587](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1730587) - Fixes #5992 - `trap` is missing as a statement to which `break` applies; `trap`-related example is flawed
Fixes [AB#1730588](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1730588) - Fixes #5994 - Warn of the pitfalls of `break` and `continue` uses outside constructs that directly support them

## PR Context

Select the type(s) of documents being changed.

**Cmdlet reference & about_ topics**
- [x] Version 7.1 preview content
- [x] Version 7.0 content
- [x] Version 6 content
- [x] Version 5.1 content

**Conceptual content**
- [ ] Fundamental conceptual articles
- [ ] Script sample articles
- [ ] DSC articles
- [ ] Gallery articles
- [ ] JEA articles
- [ ] WMF articles
- [ ] SDK articles
- [ ] Community content

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md [hunk 1]
```diff
@@ -34,13 +39,13 @@ for($i=1; $i -le 10; $i++) {
 }
 ```
 
-In this example, the `Break` statement exits the `For` loop when the `$i`
-variable equals 1. Even though the `For` statement evaluates to True until `$i`
+In this example, the `break` statement exits the `for` loop when the `$i`
+variable equals 1. Even though the `for` statement evaluates to True until `$i`
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:43]: Should the True be bold?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md [hunk 2]
```diff
@@ -34,13 +39,13 @@ for($i=1; $i -le 10; $i++) {
...
 is greater than 10, PowerShell reaches the break statement the first time the
-`For` loop is run.
+`for` loop is run.
 
-It is more common to use the Break statement in a loop where an inner condition
-must be met. Consider the following `Foreach` statement example:
+It is more common to use the `Break` statement in a loop where an inner
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:47]: Lower-case break for consistency.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md [hunk 3]
```diff
@@ -54,46 +59,23 @@ foreach ($val in $varB) {
 Write-Host "30 was found in array index $i"
 ```
 
-In this example, the `Foreach` statement iterates the `$varB` array. The `If`
+In this example, the `foreach` statement iterates the `$varB` array. The `If`
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:62]: Lower-case if for consistency.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md [hunk 4]
```diff
@@ -169,18 +155,75 @@ switch -regex ($var) {
...
-conditions are met. This example uses the `Break` statement to display results
+When PowerShell reaches the first `break` statement, the `switch` statement
+exits. If the four `break` statements are removed from the example, all four
+conditions are met. This example uses the `break` statement to display results
 when the most specific condition is met.
 
-## SEE ALSO
+## Using break in a trap statement
+
+If the final statement executed in the body of a trap statement is `break`,
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:171]: Codefence trap?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md [hunk 5]
```diff
@@ -169,18 +155,75 @@ switch -regex ($var) {
...
+  'Before loop'
+  while ($true) {
+     "1 / $i = " + (1 / $i--)
+  }
+  'After loop'
+}
+test
+```
+
+Notice that execution stop at the exception. The `After loop` is never reached.
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:194]: "execution stops"

Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:194]: Why is 'After loop' a codefence?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md [hunk 6]
```diff
@@ -169,18 +155,75 @@ switch -regex ($var) {
...
+divide by zero trapped
+Attempted to divide by zero.
+At line:10 char:6
++      "1 / $i = " + (1 / $i--)
++      ~~~~~~~~~~~~~~~~~~~~~~~~
+    + CategoryInfo          : NotSpecified: (:) [], ParentContainsErrorRecordException
+    + FullyQualifiedErrorId : RuntimeException
+```
+
+## Do not use break outside of a loop, switch or trap
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:211]: No Oxford comma? :)

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md [hunk 7]
```diff
@@ -169,18 +155,75 @@ switch -regex ($var) {
...
+
+When `break` is used outside of a construct that directly supports it
+(loops, `switch`, `trap`), PowerShell looks _up the call stack_ for an
+enclosing construct. If it can't find an enclosing construct, the current
+runspace is quietly terminated.
+
+This means that functions and scripts that inadvertently use a `break` outside
+of an enclosing construct that supports it, can inadvertently terminate their
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:219]: The comma after "supports it" isn't necessary.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md [hunk 8]
```diff
@@ -1,28 +1,31 @@
...
 ---
 # About Continue
 
-## SHORT DESCRIPTION
+## Short description
 
-Describes how the `Continue` statement immediately returns the program flow
-to the top of a program loop.
+Describes how the `continue` statement immediately returns the program flow
+to the top of a program loop, a switch statement, or a trap statement.
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md:13]: Codefence switch and trap.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md [hunk 9]
```diff
@@ -1,28 +1,31 @@
...
-[about_Break](about_Break.md).
+## Using continue in loops
 
-In the following example, program flow returns to the top of the While loop
+An unlabeled `continue` statement immediately returns the program flow to
+the top of the innermost loop that is controlled by a `For`, `Foreach`, `Do`,
+or `While` statement. The current iteration of the loop is terminated and the
+loop continues with the next iteration.
+
+In the following example, program flow returns to the top of the `While` loop
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md:28]: Shouldn't these codefenced elements be lower case to match about_Break?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md [hunk 10]
```diff
@@ -41,7 +44,7 @@ while ($ctr -lt 10)
 
 When using a `For` loop, execution continues at the `<Repeat>` statement,
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md:45]: Lower-case for.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md [hunk 11]
```diff
@@ -53,12 +56,78 @@ for ($i = 0; $i -lt 10; $i++)
...
+        $i--
     }
 }
 ```
 
-## SEE ALSO
+### Using a labeled continue in a loop
+
+A labeled `continue` statement terminates execution of the iteration and
+transfers control to the targeted enclosing iteration or switch statement
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md:67]: Codefence switch.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md [hunk 12]
```diff
@@ -53,12 +56,78 @@ for ($i = 0; $i -lt 10; $i++)
...
+## Using continue in a switch statement
+
+An unlabeled `continue` statement within a switch terminates execution of the
+current switch iteration and transfers control to the top of the switch to get
+the next input item.
+
+When there is a single input item `continue` exits the entire switch statement.
+When the switch input is a collection, the switch tests each element of the
+collection. The continue exits the current iteration and the switch continues
+with the next element.
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md:96]: Several 'switch' and 'continue' elements need codefencing in these paragraphs.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md [hunk 13]
```diff
@@ -53,12 +56,78 @@ for ($i = 0; $i -lt 10; $i++)
...
+3
+```
+
+## Using continue in a trap statement
+
+If the final statement executed in the body a `trap` statement is `continue`,
+the trapped error is silently ignored and execution continues with the
+statement immediately following the one that caused `trap` to occur.
+
+## Do not use continue outside of a loop, switch or trap
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md:116]: No Oxford comma? ;)

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Trap.md [hunk 14]
```diff
@@ -341,19 +333,22 @@ break_example
...
 
-If you include a `Continue` keyword in a `Trap` statement, PowerShell resumes
-after the statement that caused the error, just as it would without `Break` or
+If you include a `Continue` keyword in a `trap` statement, PowerShell resumes
+after the statement that caused the error, just as it would without `break` or
 `Continue`. With the `Continue` keyword, however, PowerShell does not write an
 error to the error stream.
 
-The following sample function uses the `Continue` keyword in a `Trap`
+The following sample function uses the `Continue` keyword in a `trap`
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Trap.md:351]: Lower case these continues.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Trap.md [hunk 15]
```diff
@@ -379,7 +374,7 @@ statement runs. No error is written to the error stream.
 
 ## Notes
 
-Trap statements provide a simple way to broadly ensure all terminating errors
+`trap` statements provide a simple way to broadly ensure all terminating errors
 within a scope are handled. For more finer-grained error handling, use
 `try`/`catch` blocks where traps are defined using `Catch` statements. The
 `Catch` statements only apply to the code inside the associated `Try`
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Trap.md:380]: Lower case the catch and try elements.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md [hunk 16]
```diff
@@ -169,18 +155,75 @@ switch -regex ($var) {
 ```
 
 In this example, the `$var` variable is created and initialized to a string
-value of "word2". The Switch statement uses the **Regex** class to match the
+value of `word2`. The `switch` statement uses the **Regex** class to match the
 variable value first with the term "word2". (The **Regex** class is a regular
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:159]: missed it. fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md (hunk 1 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:43]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md (hunk 2 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:47]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md (hunk 3 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:62]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md (hunk 4 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:171]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md (hunk 11 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md:67]: Fixed throughout

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md (hunk 12 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md:96]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md (hunk 13 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md:116]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Trap.md (hunk 14 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Trap.md:351]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Trap.md (hunk 15 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Trap.md:380]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md (hunk 5 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:193]: fixed

Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:193]: To make it stand out. Bold didn't seem right. And quotes are weak.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md (hunk 6 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:210]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md (hunk 7 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Break.md:218]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md (hunk 8 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md (hunk 9 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md:28]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md (hunk 10 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Continue.md:45]: Fixed

**Pull request merged**